jetty-core/jetty-websocket/jetty-websocket-jetty-tests/src/test/java/org/eclipse/jetty/websocket/tests/server/FrameListenerTest.java

```diff
@@ -149,7 +147,6 @@ public void onWebSocketFrame(Frame frame, Callback callback)
                 BufferUtil.toUTF8String(frame.getPayload()),
                 frame.getPayloadLength()));
             callback.succeed();
-            session.demand();
```

Comment:
@lachlan-roberts @lorban while currently correct, the removal of this line is not ideal.

If an application wants to handle WebSocket frames directly and only override `onWebSocketFrame()`, it has no direct control over the demand.

I opened https://github.com/jetty/jetty.project/issues/11307